config/routes.rb

```diff
@@ -5,5 +5,7 @@
   devise_for :users
   resources :products
   resources :users, only: [:index, :show, :destroy, :delete]
+  get "/dashboard" => "users#dashboard"
+  put "/products/:id/publish" => "products#publish"
```

Comment:
You don't need to do it this "manual" way.  Rails has some syntax so that under `resources :products` you can do something like this:

```
resources :products do
  member do
    put 'publish'
  end
end
```

or:

```
resources :products do
  put 'publish', on: :member
end
```

Reference: https://guides.rubyonrails.org/routing.html#adding-more-restful-actions